src/IO/HTTPCommon.cpp

```diff
@@ -91,56 +91,6 @@ namespace
         return session;
     }
 
-    class SingleEndpointHTTPSessionPool : public PoolBase<Poco::Net::HTTPClientSession>
```

Comment:
As far as I see `class SingleEndpointHTTPSessionPool ...` can be removed from .h and only stay in .cpp if we just replace 
```using PooledHTTPSessionPtr = SingleEndpointHTTPSessionPool::Entry;```
with
```using PooledHTTPSessionPtr = PoolBase<Poco::Net::HTTPClientSession>::Entry;```

Reply:
Addressed. And I leave a comment `// SingleEndpointHTTPSessionPool::Entry` after PooledHTTPSessionPtr alias to let other know it is related to the class `SingleEndpointHTTPSessionPool`